kubetest2-gke/deployer/options/build.go

```diff
@@ -17,41 +17,24 @@ limitations under the License.
 package options
 
 import (
-	"fmt"
-
 	"sigs.k8s.io/kubetest2/pkg/build"
 )
 
 type BuildOptions struct {
-	Strategy      string `flag:"~strategy" desc:"Determines the build strategy to use either make or bazel."`
-	StageLocation string `flag:"~stage" desc:"Upload binaries to gs://bucket/ci/job-suffix if set"`
-	RepoRoot      string `flag:"-"`
-	build.Builder
-	build.Stager
+	CommonBuildOptions *build.Options
 }
 
+var _ build.Builder = &BuildOptions{}
```

Comment:
Curious what the propose of this pattern is? Just type enforcement?

Reply:
yes, this essentially is a compile time type assertion

Reply:
adding: this is a common enough pattern in go, I'm not sure if it's strictly compile time or not though. 

technically I bet at minimum it also ensures that the package defining the interface is valid to import even if it otherwise wouldn't be, and importing a package can have side effects (namely packages have `init()` / globals in a binary IFF they are imported somewhere).